src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,155 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
   }
-
+  
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
     log.info("Reregistered framework.");
   }
 
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeTask(status.getTaskId());
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status); // TODO(rubbish) could terminal/running get pushed to
+                                             // this
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_1);
+          }
+          break;
+        case NAME_NODE_1 :
+          if (liveState.getNameNodeSize() == 1 && liveState.getFirstNameNodeTaskId() != null) {
```

Comment:
What if NN1 dies at the same time NN2 starts running? Can we call NN2 "FirstNameNode" and then transition to NN2 phase, or do we still need to start a NN1 as "First", and then move to NN2 phase, which may be a Noop? In that case, do we always want to `break` between these cases?